ocaml/xapi/xapi_pool.ml

```diff
@@ -1861,3 +1861,9 @@ let add_to_guest_agent_config ~__context ~self ~key ~value =
 let remove_from_guest_agent_config ~__context ~self ~key =
   Db.Pool.remove_from_guest_agent_config ~__context ~self ~key;
   Xapi_pool_helpers.apply_guest_agent_config ~__context
+
+let set_default_SR ~__context ~self ~sr =
+  Db.Pool.set_default_SR ~__context ~self ~value:sr;
+  if Db.SR.get_shared ~__context ~self:sr then begin
```

Comment:
Also, if the new default SR is shared and we have the redo log enabled on another SR already, do we want to move the redo log or should we leave it where it is?